src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDesignTimeDescriptor.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Razor.TagHelpers
+{
+    /// <summary>
+    /// A metadata class containing design time information about a tag helper.
+    /// </summary>
+    public class TagHelperDesignTimeDescriptor
+    {
+        /// <summary>
+        /// Instantiates a new instance of <see cref="TagHelperDesignTimeDescriptor"/>.
+        /// </summary>
+        /// <param name="summary">A summary on how to use a tag helper.</param>
+        /// <param name="remarks">Remarks on how to use a tag helper.</param>
+        /// <param name="outputElementHint">The HTML element a tag helper may output.</param>
+        public TagHelperDesignTimeDescriptor(string summary, string remarks, string outputElementHint)
```

Comment:
It's deffffinitely annoying. The only reason why this descriptor uses the ctor to initialize everything is because the `TagHelperDescriptor` and `TagHelperAttributeDescriptor` do; aka for consistency. 

At the initial creation of the descriptors I had them all constructorless but after a while the constructors were added and have stuck around. @dougbu If I remember correctly you showed the interest in adding the constructors initially which made sense at the time. Now that there's a boat load of properties on all of our descriptors how do you feel about nuking the ctors?

Will file an issue if @dougbu agrees (changing from ctors to no ctors will be a lot of noise + require changes to all of our comparers).